hs_core/views/resource_folder_hierarchy.py

```diff
@@ -341,6 +347,82 @@ def data_store_folder_unzip_public(request, pk, pathname):
     return data_store_folder_unzip(request, res_id=pk, zip_with_rel_path=sys_pathname)
 
 
+@api_view(['POST'])
+def data_store_add_reference(request):
+    """
+    create the reference url file, add the url file to resource, and add the url to
+    metadata accordingly for easy later retrieval
+    :param request:
+    :return: JsonResponse with status code and message
```

Comment:
Could you return the file_id in the response.  It'd be nice for the rest api.

Reply:
@sblack-usu OK. Will do.